inc/sitemaps/class-sitemaps.php

```diff
@@ -0,0 +1,432 @@
+	 * @param string $default The default value for the frequency.
+	 * @param string $url     The URL of the current entry.
+	 *
+	 * @return mixed|void
+	 */
+	static public function filter_frequency( $filter, $default, $url ) {
+		/**
+		 * Filter: 'wpseo_sitemap_' . $filter . '_change_freq' - Allow filtering of the specific change frequency
+		 *
+		 * @api string $default The default change frequency
+		 */
+		$change_freq = apply_filters( 'wpseo_sitemap_' . $filter . '_change_freq', $default, $url );
+
+		if ( ! in_array( $change_freq, array(
+			'always',
+			'hourly',
+			'daily',
+			'weekly',
+			'monthly',
+			'yearly',
+			'never',
+		) )
+		) {
+			$change_freq = $default;
+		}
+
+		return $change_freq;
+	}
+
+	/**
+	 * Spits out the XSL for the XML sitemap.
+	 *
+	 * @param string $type Type to output.
+	 *
+	 * @since 1.4.13
+	 */
+	public function xsl_output( $type ) {
+
+		if ( $type !== 'main' ) {
+
+			do_action( 'wpseo_xsl_' . $type ); // TODO document action. R.
+
+			return;
+		}
+
+		header( $this->http_protocol . ' 200 OK', true, 200 );
+		// Prevent the search engines from indexing the XML Sitemap.
+		header( 'X-Robots-Tag: noindex, follow', true );
+		header( 'Content-Type: text/xml' );
+
+		// Make the browser cache this file properly.
+		$expires = YEAR_IN_SECONDS;
+		header( 'Pragma: public' );
+		header( 'Cache-Control: maxage=' . $expires );
+		header( 'Expires: ' . gmdate( 'D, d M Y H:i:s', ( time() + $expires ) ) . ' GMT' );
+
+		require_once( WPSEO_PATH . 'css/xml-sitemap-xsl.php' );
+	}
+
+	/**
+	 * Spit out the generated sitemap and relevant headers and encoding information.
+	 */
+	public function output() {
+
+		if ( ! headers_sent() ) {
+			header( $this->http_protocol . ' 200 OK', true, 200 );
+			// Prevent the search engines from indexing the XML Sitemap.
+			header( 'X-Robots-Tag: noindex, follow', true );
+			header( 'Content-Type: text/xml' );
+		}
+
+		echo $this->renderer->get_output( $this->sitemap, $this->transient );
+	}
+
+	/**
+	 * Build the `<url>` tag for a given URL.
+	 *
+	 * @deprecated
+	 *
+	 * @param array $url Array of parts that make up this entry.
+	 *
+	 * @return string
+	 */
+	public function sitemap_url( $url ) {
+
+		return $this->renderer->sitemap_url( $url );
+	}
+
+	/**
+	 * Make a request for the sitemap index so as to cache it before the arrival of the search engines.
+	 */
+	public function hit_sitemap_index() {
+		wp_remote_get( wpseo_xml_sitemaps_base_url( 'sitemap_index.xml' ) );
+	}
+
+	/**
+	 * Get the GMT modification date for the last modified post in the post type.
+	 *
+	 * @param string|array $post_types Post type or array of types.
+	 *
+	 * @return string|false
+	 */
+	static public function get_last_modified_gmt( $post_types ) {
+
+		global $wpdb;
+
+		static $post_type_dates = array();
+
+		if ( ! is_array( $post_types ) ) {
+			$post_types = array( $post_types );
+		}
+
+		if ( empty( $post_type_dates ) ) {
+
+			$post_type_dates = array();
```

Comment:
This is already declared as array. I guess this isn't necessary. Please correct me if I am wrong.